examples/aggregator/module/rust/src/application.rs

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::aggregation::{Aggregation, Monoid};
+use crate::proto::aggregator::Vector;
+use crate::proto::aggregator_grpc::{dispatch, Aggregator};
+use log::{info, warning};
+use oak::grpc;
+use protobuf::well_known_types::Empty;
+use std::fmt::Debug;
+
+pub trait Serializable<T> {
+    fn deserialize(proto: &T) -> Self;
+    fn serialize(&self) -> T;
+}
+
+/// Oak Application that collects aggregated data.
+pub struct Application<T: Monoid> {
```

Comment:
I don't see the point of the generic parameter here. I think `Application` should be a concrete struct, and just refer to `Vector` directly. The reusable part of this crate should just be the `Aggregation` trait and its dependencies, not the application (or node) itself.

Reply:
Done